app/controllers/machine_conditions_controller.rb

```diff
@@ -12,22 +12,26 @@ def create
   end
 
   def destroy
+    user = current_user.nil? ? nil : current_user
     mcx = MachineCondition.find(params[:id])
-    mcx.destroy
+
+    mcx.destroy if user && (user.id == mcx.user_id)
 
     render nothing: true
   end
 
   def update
```

Comment:
turns out the update method never gets used... probably should comment it out or delete it.

Reply:
You meant the create, right? If so, I zapped it out.